### PR TITLE
Remove some `NodePath` methods

### LINKED ISSUE #16443: Cleanup `NodePath` methods
There are a lot of methods on `NodePath` that aren't really meant to be public -- it's just that we expose all the function exported by the internal files that implement the various `NodePath` actually-public methods.

We should go through all of them and only expose those that are actually used outside of `@babel/traverse`, ideally with an explicit list rather than doing 
https://github.com/babel/babel/blob/941cee23d861f9bb8cc0e08dbbb30da5be047190/packages/babel-traverse/src/path/index.ts#L220-L233

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #16443 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | https://github.com/babel/website/pull/2924
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

File: packages/babel-traverse/src/path/index.ts [hunk 1]
```diff
@@ -302,21 +289,43 @@ const methods = {
...
+
   // @ts-expect-error babel 7 only
   NodePath_Final.prototype.arrowFunctionToShadowed =
-    // workaround for rollup
     // @ts-expect-error babel 7 only
     NodePath_conversion[String("arrowFunctionToShadowed")];
+
+  Object.assign(NodePath_Final.prototype, {
+    // @ts-expect-error Babel 7 only
+    has: NodePath_introspection[String("has")],
```
Comment (nicolo-ribaudo) [inline packages/babel-traverse/src/path/index.ts:301]: What is `String(...)` for?

File: packages/babel-traverse/src/path/modification.ts [hunk 2]
```diff
@@ -399,14 +399,17 @@ export function pushContainer<
...
-) {
-  const hoister = new PathHoister<T>(this, scope);
-  return hoister.run();
+if (!process.env.BABEL_8_BREAKING && !USE_ESM) {
+  /**
+   * Hoist the current node to the highest scope possible and return a UID
+   * referencing it.
+   */
+  // eslint-disable-next-line no-restricted-globals
+  exports.hoist = function hoist<T extends t.Node>(
```
Comment (nicolo-ribaudo) [inline packages/babel-traverse/src/path/modification.ts:408]: Out of curiosity, what was this ever used for?

File: packages/babel-traverse/src/path/index.ts (hunk 1 above)
Comment (liuxingbaoyu) [reply packages/babel-traverse/src/path/index.ts:301]: This is for rollup, otherwise it will show that the export cannot be found.

Comment (nicolo-ribaudo) [reply packages/babel-traverse/src/path/index.ts:301]: Can you add a comment? :)

File: packages/babel-traverse/src/path/modification.ts (hunk 2 above)
Comment (liuxingbaoyu) [reply packages/babel-traverse/src/path/modification.ts:408]: It seems to be used by `babel-plugin-transform-react-constant-elements`, I'm not sure, it has no use now.

**Review by nicolo-ribaudo — APPROVED** [review]

**Comment by kungfooman** [comment]
There is no reasoning here why the change is needed, can someone enlighten the rest of us? To me this just starts looking more complicated and unidiomatic code now, for example:

```diff
- path.resync();
+ resync.call(path);
```

It looks like functional programming is trying to take over and making it unreadable.

**Comment by liuxingbaoyu** [comment]
@kungfooman This is to make them private and users won't be able to call them. We may avoid passing `this` in the future.

**Comment by nicolo-ribaudo** [comment]
Yeah, in the future that will just become `resync(path)`.

**Review by JLHwung — APPROVED** [review]

**Pull request merged**